neuralpp/test/quick_tests/symbolic/normalizer_test.py

```diff
@@ -214,11 +218,65 @@ def test_quantifier_normalizer():
 
     expr = f(if_then_else(A, B, C), BasicSummation(int, D, empty_context & (0 < D) & (D < 10),
                                                    B * if_then_else(B == 4, B + C, 1)))
+    product2 = SymPyExpression.from_sympy_object(144 + 36 * CC, {CC: int})
     assert normalizer.normalize(expr, empty_context).syntactic_eq(
         if_then_else(A,
-                     if_then_else(B == 4,
-                                  f(4, BasicSummation(int, D, empty_context & (0 < D) & (D < 10), 16 + 4 * C)),
-                                  f(B, BasicSummation(int, D, empty_context & (0 < D) & (D < 10), B))),
-                     if_then_else(B == 4,
-                                  f(C, BasicSummation(int, D, empty_context & (0 < D) & (D < 10), 16 + 4 * C)),
-                                  f(C, BasicSummation(int, D, empty_context & (0 < D) & (D < 10), B)))))
+                     if_then_else(B == 4, f(4, product2), f(B, 9 * B)),
+                     if_then_else(B == 4, f(C, product2), f(C, 9 * B))))
+
+
+def test_quantifier_normalizer_integration():
+    i = BasicVariable('i', int)
+    empty_context = Z3SolverExpression()
+    i_range = empty_context & (0 < i) & (i < 10)
+    integral = BasicIntegral(i, i_range, i)
+
+    context = empty_context
+    normalizer = GeneralNormalizer()
+    # 1/2 * i ** 2
+    assert normalizer.normalize(integral, context).syntactic_eq(BasicConstant(50))
+    # 1/2 * i ** 2 + i
+    assert normalizer.normalize(BasicIntegral(i, i_range, i + 1), context).syntactic_eq(BasicConstant(60))
+
+    #          *
+    #       /    \
+    #     if A   Integral
+    #     /  \    |
+    #    B    C  B * if B>4
+    #                /    \
+    #               B+C   Integral
+    #                       |
+    #                      if B<5
+    #                      / \
+    #                     C   1
+    # should be normalized to
+    #              if A
+    #           /        \
+    #      if B>4       if B>4
+    #       /    \        /   \
+    #      *      ..   ..      * (= C * 10 * 50 = 500 * B * C)
+    #     /\                  /  \
+    #    B  Integral(D)      C   Integral(D)    (= *10)
```

Comment:
I am having a hard time matching this diagram and the actual expressions in the program. For example, the diagram shows `Integral(C)` in the last line 261 without a body, and in that same line we see `50 * B` but the code has `500*B` instead. Line 259 includes an expression `*10` (is this a product of a single number?).

Reply:
Added comments explaining that